.github/workflows/testing-and-deployment.yml

```diff
@@ -166,6 +166,11 @@ jobs:
           restore-keys: |
             Examples-1-
 
+      - name: Setup xvfb dependencies
+        run: |
+          git clone --depth=1 https://github.com/mne-tools/mne-python.git --branch main --single-branch
+          ./mne-python/tools/setup_xvfb.sh
+
```

Comment:
Woah... why was this added here? This needs to be removed ASAP as this section of the tests is strictly non-graphics.

Reply:
Further I want to discourage the use of custom scripts from external projects in our core CI routines. We should keep any configuration and set up limited to well-established GitHub Actions or scripts managed by PyVista.

Adding and executing arbitrary shell scripts is a major red flag and security risk albeit this is from a well-trusted downstream project.

Reply:
That sounds reasonable. They were added for PyQtx.QWidgets.QApplication.process_events testing. Not sure there is a way to fake that for non-graphics tests, I just don't know there might be though. Otherwise since it was tested it could be no covered.